File: trabalhos 1 ANO/rio.js
class Rio {
    constructor(nome, nivel, poluido){
        this.nome = nome
        this.nivel = nivel
        this.poluido = poluido
    }

    chover = (x) => {
        this.nivel += x
    }
    ensolarar = (x) => {
        if(this.nivel - x >= 0){
            this.nivel -= x
        }
        else{
            console.log(`Impossivel deixar o nivel do rio abaixo de zero`)
        }
    }
    limpar = () => {
        this.poluido = false
    }
    sujar = () => {
        this.poluido = true
    }
    mostra = () => {
        let aux = this.poluido ? "Sujo" : "Limpo"
        console.log(`Nome : ${this.nome}, Nivel : ${this.nivel} e esta ${aux}`)
    }
}

File: NETBEANS_PROJECTS_JV/Lista1/src/main/java/br/com/unifacef/lista1/ex4/Rio.java

package br.com.unifacef.lista1.ex4;

public class Rio {
    String nome;
    float nivel;
    boolean poluido;
    
    public Rio(){
    }
    
    public Rio(String nome, float nivel, boolean poluido){
        this.nome = nome;
        this.nivel = nivel;
        this.poluido = poluido;        
    }
    
    public void chover(float x){
        this.nivel = this.nivel + x;
    }
    
    public void ensolarar(float x){
        if (this.nivel - x >= 0){  
                this.nivel = this.nivel - x;
            }
            else {
                System.out.println("Impossivel, nivel ficara negativo! \n");
            }
    }
    
    public void limpar(){
        if (this.poluido == false){
               this.poluido = true;
           }
           else {
            System.out.println("Impossivel, rio ja esta limpo");
           }
    } 
    
    public void sujar(){
        if (this.poluido == true){
               this.poluido = false;
           }
           else {
            System.out.println("Impossivel, rio ja esta sujo");
           }
    }
    
    public String mostra(){
        return " Nome do rio : " + this.nome + "\n Nivel : " + this.nivel + "\n Puluido : " + (this.poluido ? "Sim" : "Não") + "\n";
    }
}
